server/manager/src/main/java/org/apache/accumulo/manager/tableOps/delete/DeleteTable.java

```diff
@@ -41,6 +42,19 @@ public DeleteTable(NamespaceId namespaceId, TableId tableId) {
 
   @Override
   public long isReady(long tid, Manager env) throws Exception {
+
+    // Before attempting to delete the table, cancel any running user
+    // compactions.
+    if (Utils.reserveNamespace(env, namespaceId, tid, false, true, TableOperation.COMPACT_CANCEL)
```

Comment:
@dlmarion did you look into the behavior for the following situation :

 1. gets read lock
 2. cancels
 3. release read lock
 4. get write lock
 5. manager process dies
 6. manager process starts up again
 7. isReady is called again and it attempts to get the read lock again

When step 7 runs and tries to get the read lock, the FATE tx will already be holding the write lock because of the failure. What will happen in this case?

Reply:
I did not look into that behavior, I just implemented @ctubbsii  suggestion about moving the logic server-side. I don't know enough about FATE to even know that issue would occur. Is there a way to test if this FATE transaction has a read or write lock held already?

Reply:
Each FATE Repo has to be able to fail at any point and be re-run.   As for the FATE lock I have no idea what the behavior would be, I have not looked at the code in a really long time.  Would just need to examine the code to see what it might do.  Also I do see some sort of unit test, maybe it could be used for experimentation.

Reply:
We could make sure the lock ID used for the read is the same as that used by the delete write lock. I would expect that one holding the write lock could also get a read lock, since that's the normal behavior of ReadWriteLock. But you can't upgrade a read lock to a write lock.

Reply:
I think I can easily add a method to check if the Fate transaction already has the write lock. I should have something later this afternoon.

Reply:
ok, this is going to take longer than I thought. I'm trying to understand how the locks work. Do you want me to revert the commit?

Reply:
Reverting or opening an issue that is a blocker for 2.1 seem like two good ways to handle it.

Reply:
I think I might have it. I'm going to open another PR here shortly

Reply:
PR up at https://github.com/apache/accumulo/pull/2175